src/content/docs/browser/browser-monitoring/tutorials/combining-npm-nerdgraph.mdx

```diff
@@ -0,0 +1,166 @@
+  src={nerdgraphExplorerBrowserApplicationConfig}
+/>
+
+<Callout variant="tip">
+Need more information on using the NerdGraph APIs or Explorer? [Checkout the NerdGraph Introduction page](/docs/apis/nerdgraph/get-started/introduction-new-relic-nerdgraph/).
+</Callout>
+
+</Step>
+
+<Step>
+### Install the config
+
+Copy the contents of the `jsConfig` property and replace the contents of the `options` object your source file.
+</Step>
+</Steps>
+
+## Depoloying using the npm package
+
+With the npm package installed and properly configured, you can now build and deploy your browser application. Load your website in a browser and verify there are network calls being made to the `beacon` URI from the config.
+
+## Inject browser application config into HTML
+
+As an alternative to placing the browser application configuration in your source code, you may want to inject the configuration into the HTML of your website. This is especially useful when your source code is reused across multiple websites.
+
+<Steps>
+<Step>
+### Modify source code snippet
+
+Modify the first snippet in your source code to `import` and instantiate the `BrowserAgent` class with no parameters.
+
+```javascript
+import { BrowserAgent } from '@newrelic/browser-agent/loaders/browser-agent'
+// Remaining import statements
+
+// The agent loader code executes immediately on instantiation.
+new BrowserAgent()
+
+// Remaining code
+```
+</Step>
+
+<Step>
+### Modify the graphql query
+
+Modify the graphql query to retrieve the browser application config as a JSON encoded script string. Run the new query remembering to replace `<guid>` with the browser application entity guid retrieved earlier.
+
+```graphql
+query {
+  actor {
+    entity(guid: '<guid>') {
+      ... on BrowserApplicationEntity {
+        guid
+        name
+        browserProperties {
+          jsConfigScript
+        }
+      }
+    }
+  }
+}
+```
+</Step>
+
+<Step>
+### Install the config
+
+Copy the contents of the `jsConfigScript` property. You will need to run the results through a JSON parser or replace the values of `\"` with `"`. The result will be a string containing a `<script>` element with all the browser application config information. Place the `<script>` element at the top of the `<head>` element, but below any `<meta>` elements, of your website HTML.
+</Step>
+</Steps>
+
+## Taking it further
+
+This tutorial only scratches the surface of how you might use the NerdGraph APIs and the browser agent npm package. Some ideas for further integration:
+
+- Integrate the browser agent into your website builder application and use the NerdGraph APIs to retrieve the browser application config for each of your customers when they enable New Relic monitoring within your management dashboard.
+- Create a shared browser agent component in your organization's component library and automatically injects the configuration into the HTML using an CDN edge function that calls the NerdGraph API.
```

Comment:
```suggestion
- Create a generic shared browser agent component in your organization's component library using the npm package, then automatically inject a configuration block into the HTML using a CDN edge function that calls the NerdGraph API.
```

Reply:
I'm not that experienced with edge functions, but I think the implication here is that the edge function would be responsible for determining which config block to use based on where the app is served, which application the component is part of, or other relevant variables. Can an edge function be deterministic and cache output based on inputs? Otherwise might be a little spammy on the NerdGraph queries. I sense that some variation of what you're describing could be very cool. Just curious how it could work.

Reply:
Depends on the service and architecture. Akamai has EdgeKV which can be used to store the NG response and recall it.